_books/ion-1-1/src/binary/template_placeholders.md

```diff
@@ -0,0 +1,102 @@
+# Template Placeholders
+
+> [!NOTE]
+> This chapter focuses on the binary encoding of template placeholders.
+> The [_Macros_](../macros.md) section explains what they are and how they are used.
+
+Template placeholders are special constructs used within macro template bodies to indicate where macro arguments should be substituted.
+They are encoded using opcodes `0xE8`, `0xE9`, and `0xEA`.
+
+Placeholders may only occur in _value_ position, only within a macro body; they are illegal anywhere else.
+
+### Placeholder Types
+
+| Opcode | Placeholder Type                           |
+|--------|--------------------------------------------|
+| `0xE8` | Tagged template placeholder (no default)   |
+| `0xE9` | Tagged template placeholder (with default) |
+| `0xEA` | Tagless template placeholder               |
+
+### Tagged Template Placeholder with No Default
+
+Opcode `0xE8` indicates a tagged template placeholder with no default value. No additional bytes follow.
+
+##### Encoding of `(:?)`
+```
+┌──── Opcode 0xE8 indicates a tagged template placeholder with no default
+E8
+```
+
+##### Encoding of `foo::(:?)`
+```
+      ┌── Annotation text: foo
+┌─────┴──────┐ ┌──── Opcode 0xE8 indicates a tagged template placeholder with no default
+59 07 66 6F 6F E8
+```
+
+### Tagged Template Placeholder with Default
+
+Opcode `0xE9` indicates a tagged template placeholder with a default value.
+The default value follows and may be any value or e-expression that produces a value.
+A `NOP` is legal, and ignored.
```

Comment:
Let's only do this if allowing it results in simpler code than not allowing it.

Reply:
I think it does result in simpler code. Any time we see a `NOP`, I think it's easier to just ignore it than it is to check and see whether it's legal in that context. (FWIW, I would even be okay with allowing `NOP` between and after annotations with the expectation that there must eventually be a non-`NOP` opcode that produces a value.)